### PR TITLE
refactor: centralize keyboard shortcuts and fix bold shortcut conflicts

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

- Move keyboard shortcuts (Cmd+B for navigation, Cmd+/ for new conversation) to centralized hook
- Disable default Cmd+B shortcut from Tiptap's Bold extension

## Risk

N/A

## Deploy Plan

Deploy front

### CODE REVIEW

File: front/components/assistant/conversation/input_bar/editor/useCustomEditor.tsx [hunk 1]
```diff
@@ -156,6 +157,14 @@ export interface CustomEditorProps {
   disableAutoFocus: boolean;
 }
 
+const CustomBold = Bold.extend({
```
Comment (flvndvd) [inline front/components/assistant/conversation/input_bar/editor/useCustomEditor.tsx:160]: Hacker 🙃 !

File: front/components/navigation/Navigation.tsx [hunk 2]
```diff
@@ -18,6 +18,8 @@ interface NavigationProps {
   subscription: SubscriptionType;
   navChildren?: React.ReactNode;
   subNavigation?: SidebarNavigation[] | null;
+  isNavigationBarOpen: boolean;
```
Comment (flvndvd) [inline front/components/navigation/Navigation.tsx:21]: Why not using the Sidebar context for this?

File: front/hooks/useAppKeyboardShortcuts.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import type { LightWorkspaceType } from "@dust-tt/types";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+
+export function useAppKeyboardShortcuts(owner: LightWorkspaceType) {
+  const [isNavigationBarOpen, setIsNavigationBarOpen] = useState(true);
+  const router = useRouter();
+
+  useEffect(() => {
+    function handleKeyboardShortcuts(event: KeyboardEvent) {
```
Comment (flvndvd) [inline front/hooks/useAppKeyboardShortcuts.ts:10]: `nit` let's use arrow functions when nested within function's code.

File: front/components/assistant/conversation/input_bar/editor/useCustomEditor.tsx (hunk 1 above)
Comment (fontanierh) [reply front/components/assistant/conversation/input_bar/editor/useCustomEditor.tsx:160]: This is the only (and recommended) way to do this on tiptap

File: front/components/navigation/Navigation.tsx (hunk 2 above)
Comment (fontanierh) [reply front/components/navigation/Navigation.tsx:21]: Sidebar and navigation bar are not the same thing. Sidebar context doesn't do anything for the navigation bar

File: front/hooks/useAppKeyboardShortcuts.ts (hunk 3 above)
Comment (fontanierh) [reply front/hooks/useAppKeyboardShortcuts.ts:10]: Why ?

Comment (flvndvd) [reply front/hooks/useAppKeyboardShortcuts.ts:10]: Suggested as it helps improve reading/parsing the code when `function` keyword are reserved for top level functions. Anyway, it's a nit!

File: front/components/navigation/Navigation.tsx (hunk 2 above)
Comment (flvndvd) [reply front/components/navigation/Navigation.tsx:21]: Interesseting 🤯

Comment (flvndvd) [reply front/components/navigation/Navigation.tsx:21]: Ok Indeed the context is only used on Mobile 🤯

**Review by flvndvd — APPROVED** [review]

**Pull request merged**